File: Web/RMA-Web-Shell/projects/fincare/src/app/billing-manager/models/credit-note-reversal.ts
import { Transaction } from './transaction';
import { DocumentsRequest } from 'projects/shared-components-lib/src/lib/document-management/documents-request';
import { FinPayee } from '../../shared/models/finpayee';
import { Note } from 'projects/shared-components-lib/src/lib/notes/note';
import { PeriodStatusEnum } from 'projects/shared-models-lib/src/lib/enums/period-status-enum';

export class CreditNoteReversal {
  debtorAccount: FinPayee;
  rolePlayerId: number;
  finPayeNumber: string;
  isAuthorised: Date;
  authorisedBy: string;
  authorisedDate: Date;
  transactions: Transaction[];
  notes: Note[];
  amount: number;
  documents: DocumentsRequest[];
  requestCode: string;
  periodStatus: PeriodStatusEnum;
}

File: Service/Billing/RMA.Service.Billing.Contracts/Entities/CreditNoteReversal.cs
using RMA.Common.Entities;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

using System;
using System.Collections.Generic;

namespace RMA.Service.Billing.Contracts.Entities
{
    public class CreditNoteReversals
    {
        public int RolePlayerId { get; set; }
        public FinPayee DebtorAccount { get; set; }
        public string FinPayeeNumber { get; set; }
        public bool IsAuthorised { get; set; }
        public string AuthorisedBy { get; set; }
        public DateTime? AuthorisedDate { get; set; }
        public List<Transaction> Transactions { get; set; }
        public List<Note> Notes { get; set; }
        public string RequestCode { get; set; }
        public bool IsPaymentReAllocation { get; set; }
        public string ReAllocationReceiverFinPayeeNumber { get; set; }
        public bool IsCreditNoteReAllocation { get; set; }
        public PeriodStatusEnum PeriodStatus { get; set; } = PeriodStatusEnum.Current;
        public List<InvoiceAllocation> InvoiceAllocations { get; set; } = new List<InvoiceAllocation>();
        public int? ToRolePlayerId { get; set; }
    }
}
